bundles/org.openhab.core.compat1x/src/main/java/org/openhab/core/autoupdate/internal/AutoUpdateProviderDelegate.java

```diff
@@ -20,26 +37,185 @@
+
+    @Activate
+    protected void activate() {
+        refreshItemUpdateVetos();
+        started = true;
+        itemRegistry.addRegistryChangeListener(this);
+        for (org.openhab.core.autoupdate.AutoUpdateBindingProvider provider : providers) {
+            provider.addBindingChangeListener(this);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate() {
+        for (org.openhab.core.autoupdate.AutoUpdateBindingProvider provider : providers) {
+            provider.removeBindingChangeListener(this);
+        }
+        itemRegistry.removeRegistryChangeListener(this);
+        started = false;
+    }
 
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE)
     public void addAutoUpdateBindingProvider(org.openhab.core.autoupdate.AutoUpdateBindingProvider provider) {
         providers.add(provider);
+        if (started) {
+            refreshItemUpdateVetos();
+            provider.addBindingChangeListener(this);
+        }
     }
 
     public void removeAutoUpdateBindingProvider(org.openhab.core.autoupdate.AutoUpdateBindingProvider provider) {
         providers.remove(provider);
+        if (started) {
+            refreshItemUpdateVetos();
+            provider.removeBindingChangeListener(this);
+        }
+    }
+
+    @Reference
+    protected void setItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = itemRegistry;
+    }
+
+    protected void unsetItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = null;
     }
 
     @Override
-    public Boolean autoUpdate(String itemName) {
-        for (org.openhab.core.autoupdate.AutoUpdateBindingProvider provider : providers) {
-            Boolean autoUpdate = provider.autoUpdate(itemName);
-            if (autoUpdate != null) {
-                return autoUpdate;
+    public void addProviderChangeListener(ProviderChangeListener<Metadata> listener) {
+        this.listeners.add(listener);
+    }
+
+    @Override
+    public Collection<Metadata> getAll() {
+        Set<Metadata> metadataSet = new HashSet<>();
+        for (Item item : itemRegistry.getAll()) {
+            synchronized (itemUpdateVetos) {
+                if (itemUpdateVetos.contains(item.getName())) {
+                    Metadata metadata = getMetadata(item.getName());
+                    metadataSet.add(metadata);
+                }
+            }
+        }
+        return metadataSet;
+    }
+
+    @Override
+    public void removeProviderChangeListener(ProviderChangeListener<Metadata> listener) {
+        this.listeners.remove(listener);
+    }
+
+    private void refreshItemUpdateVetos() {
+        Set<String> newVetos = new HashSet<>();
+        synchronized (itemUpdateVetos) {
+            itemUpdateVetos.clear();
+            for (Item item : itemRegistry.getAll()) {
+                for (org.openhab.core.autoupdate.AutoUpdateBindingProvider provider : providers) {
+                    Boolean autoUpdate = provider.autoUpdate(item.getName());
```

Comment:
Does it really return a `Boolean` with potential null's?

Reply:
Yes. A provider returns null if it does not want to do any statement.